### PR TITLE
feat(spinner): refactor spinner to not use svg and make use of design tokens

### PR DESCRIPTION
**Pull request by mivaylo**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by mivaylo** [comment]
CDE-604

File: projects/angular/src/progress/spinner/STYLES.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Spinner
 
+## CSS Custom Properties
+
+| CSS Custom Property                    | Description                                  |
+| -------------------------------------- | -------------------------------------------- |
+| --clr-spinner-default-size             | Size (diameter) of large spinner             |
+| --clr-spinner-medium-size              | Size of medium spinner                       |
+| --clr-spinner-small-size               | Size of small spinner                        |
+| --clr-spinner-spacing-to-text          | Spacing between spinner and text next to it  |
```
Comment (valentin-mladenov) [inline projects/angular/src/progress/spinner/STYLES.md:10]: Can you rename this to `--clr-spinner-margin-right`?
There may be anything on the right not only text.

File: projects/angular/src/progress/spinner/_properties.spinner.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  :root {
+    --clr-spinner-default-size: #{variables.$clr_baselineRem_3};
+    --clr-spinner-small-size: #{variables.$clr_baselineRem_0_75};
+    --clr-spinner-medium-size: #{variables.$clr_baselineRem_1_5};
+    --clr-spinner-inside-button-size: #{variables.$clr_baselineRem_0_65};
+    --clr-spinner-inside-small-button-size: #{variables.$clr_baselineRem_0_5416};
+    --clr-spinner-default-stroke-width: #{variables.$clr_baselineRem_0_25};
+    --clr-spinner-small-stroke-width: #{variables.$clr_baselineRem_0_1};
+    --clr-spinner-medium-stroke-width: #{variables.$clr_baselineRem_0_125};
+  }
```
Comment (valentin-mladenov) [inline projects/angular/src/progress/spinner/_properties.spinner.scss:19]: Add those as properties.

```suggestion
    --clr-spinner-fill-bg-color: rgba(#{spinner-variables.$clr-spinner-bg-color}, #{spinner-variables.$clr-spinner-opacity});
    --clr-spinner-fill-inverse-bg-color: : rgba(#{spinner-variables.$clr-spinner-inverse-bg-color}, 0.15);
    --clr-spinner-fill-color: #{spinner-variables.$clr-spinner-color};
    --clr-spinner-fill-inverse-color: #{spinner-variables.$clr-spinner-inverse-color};
  }
```

File: projects/angular/src/progress/spinner/_spinner.clarity.scss [hunk 3]
```diff
@@ -13,58 +13,186 @@
...
+      border-style: solid;
+      @include mixins.css-var(
+        border-color,
+        clr-spinner-fill-snake-color,
+        spinner-variables.$clr-spinner-color,
+        variables.$clr-use-custom-properties
+      );
+      border-right-color: transparent !important;
+      border-bottom-color: transparent !important;
+      border-radius: 50%;
```
Comment (valentin-mladenov) [inline projects/angular/src/progress/spinner/_spinner.clarity.scss:54]: This should be extracted as a property token. 
```suggestion
      @include mixins.css-var(
        border-radius,
        clr-spinner-radius,
        50%,
        variables.$clr-use-custom-properties
      );
```
Add it in properties and shim as:
Prop: `--clr-spinner-radius: 50%;` 
Shim: `--clr-spinner-radius: var(--cds-alias-object-border-radius-300);`

File: projects/angular/src/progress/spinner/_spinner.clarity.scss [hunk 4]
```diff
@@ -13,58 +13,186 @@
...
+        variables.$clr-use-custom-properties
+      );
+      border-style: solid;
+      @include mixins.css-var(
+        border-color,
+        clr-spinner-fill-bg-color,
+        rgba(spinner-variables.$clr-spinner-bg-color, spinner-variables.$clr-spinner-opacity),
+        variables.$clr-use-custom-properties
+      );
+      border-radius: 50%;
```
Comment (valentin-mladenov) [inline projects/angular/src/progress/spinner/_spinner.clarity.scss:80]: This should be extracted as a property token. Same as previous comment.
```suggestion
      @include mixins.css-var(
        border-radius,
        clr-spinner-radius,
        50%,
        variables.$clr-use-custom-properties
      );
```
Add it in properties and shim as:
Prop: `--clr-spinner-radius: 50%;` 
Shim: `--clr-spinner-radius: var(--cds-alias-object-border-radius-300);`

File: projects/angular/src/progress/spinner/_spinner.clarity.scss [hunk 5]
```diff
@@ -13,58 +13,186 @@
...
+      @include mixins.min-equilateral-var(
+        clr-spinner-small-size,
+        variables.$clr_baselineRem_0_75,
+        variables.$clr-use-custom-properties
+      );
     }
 
     &.spinner-inline {
       vertical-align: text-bottom; //To align with text properly
+      margin-right: var(--clr-spinner-spacing-to-text);
```
Comment (valentin-mladenov) [inline projects/angular/src/progress/spinner/_spinner.clarity.scss:130]: Use it with mixins.
```suggestion      
      @include mixins.css-var(
        margin-right,
        clr-spinner-spacing-to-text,
        0,
        variables.$clr-use-custom-properties
      );
```

**Review by valentin-mladenov — CHANGES_REQUESTED** [review]
Check my comments.

**Pull request merged**